content/ember/v5/deprecate-array-prototype-extensions.md

```diff
@@ -0,0 +1,622 @@
+---
+id: deprecate-array-prototype-extensions
+title: "Deprecate array prototype extensions"
+until: '6.0.0'
+since: '5.0.0'
+---
+
+Ember historically extended the prototypes of native Javascript arrays to implement `Ember.Enumerable`, `Ember.MutableEnumerable`, `Ember.MutableArray`, `Ember.Array`. As of v5, the usages of array prototype extensions are deprecated.
+
+For convenient functions like `filterBy`, `compact`, you can directly convert to use native array methods.
+
+For mutation functions (like `pushObject`, `replace`) or observable properties (`firstObject`, `lastObject`), in order to keep the reactivity, you should take following steps:
+1. convert the array either to a new `@tracked` property, or use `TrackedArray` from `tracked-built-ins`;
+2. use array native methods;
+3. fully test to make sure the reactivity is maintained.
+
+### Convenient Functions
+For convenient functions like `filterBy`, `compact`, you can directly convert to use native array methods. This includes following (a list from [`EmberArray` methods](https://api.emberjs.com/ember/release/classes/EmberArray)):
+
+* `any`
```

Comment:
Rather than I would suggest doing this on a per-item basis, rather than just listing them all and then showing a large before/after. That will make it easier for people to see how to migrate any individual usage.

Reply:
Done!